sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestStream.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableList;
+
+import org.jdbi.v3.H2DatabaseRule;
+import org.jdbi.v3.Something;
+import org.jdbi.v3.sqlobject.customizers.Mapper;
+import org.jdbi.v3.sqlobject.mixins.CloseMe;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class TestStream
+{
+    @Rule
+    public H2DatabaseRule db = new H2DatabaseRule();
+
+    @Test
+    public void testReturnStream() throws Exception {
+        Something one = new Something(3, "foo");
+        Something two = new Something(4, "bar");
+        Something thr = new Something(5, "baz");
+
+        Spiffy dao = SqlObjectBuilder.onDemand(db.getDbi(), Spiffy.class);
+        dao.insert(one);
+        dao.insert(thr);
+        dao.insert(two);
+
+        List<Something> results = dao.stream().collect(Collectors.toList());
```

Comment:
`collect` is not documented to close the stream. The `close` method suggests wrapping streams in try-with-resource blocks.